or-model/src/main/java/com/google/refine/expr/WrappedRow.java

```diff
@@ -58,13 +58,9 @@ public Object getField(String name, Properties bindings) {
             return new CellTuple(project, row);
         } else if ("index".equals(name)) {
             return rowIndex;
-        } else if ("record".equals(name)) {
-            int rowIndex = (Integer) bindings.get("rowIndex");
-            
+        } else if ("record".equals(name)) {           
             return new WrappedRecord(project.recordModel.getRecordOfRow(rowIndex));
-        } else if ("columnNames".equals(name)) {
-            Project project = (Project) bindings.get("project");
-            
+        } else if ("columnNames".equals(name)) {           
```

Comment:
That's something I'll look into more closely. There are bigger changes about the cross function later on, also because we did a refactoring of it in the master branch since then.